Add jest mock for 'PushNotificationManager'

ZauberNerd opened the pull request:
`PushNotificationIOS` is a library that makes use of the
`PushNotificationManager` native module.
This commit adds a jest mock for its native module so that code that
uses `PushNotificationIOS` can be tested using jest.

In order to enable behaviour in unit tests it is possible to replace or
overwrite the mock implementation, see the [jest guide on
mocks](https://facebook.github.io/jest/docs/mock-functions.html).
By doing something similar to:
```javascript
import { NativeModules } from 'react-native';
const { PushNotificationManager } = NativeModules;
// mock 'checkPermissions' to return a different value than the default:
PushNotificationManager.checkPermissions.mockImplementationOnce((callback) => {
    // execute callback in next tick to enable async behaviour
    process.nextTick(() => callback({
        alert: false,
        badge: false,
        sound: false
    }));
});
// execute unit tests for code that makes use of 'PushNotificationIOS'
```

peterp commented on jest/setup.js:
```diff
@@ -136,6 +136,24 @@ const mockNativeModules = {
+    getApplicationIconBadgeNumber: jest.fn(callback => process.nextTick(() => callback(0))),
+    cancelLocalNotifications: jest.fn(),
+    getScheduledLocalNotifications: jest.fn(callback => process.nextTick(() => callback())),
+    requestPermissions: jest.fn(() => ({ alert: true, badge: true, sound: true })),
```
This should return a promise.

peterp replied:
This should return a promise.

ZauberNerd replied:
Good catch, thank you!
Fixed and rebased.

ZauberNerd commented:
/cc @cpojer we briefly talked about this at the JSConf.eu where you asked me to ping you on this issue.
Would love to get some feedback on this if you have the time.

cpojer commented on jest/setup.js:
```diff
@@ -156,6 +156,24 @@ const mockNativeModules = {
+    getApplicationIconBadgeNumber: jest.fn(callback => process.nextTick(() => callback(0))),
+    cancelLocalNotifications: jest.fn(),
+    getScheduledLocalNotifications: jest.fn(callback => process.nextTick(() => callback())),
+    requestPermissions: jest.fn(() => Promise.resolve({ alert: true, badge: true, sound: true })),
```
Code style: do not use spaces around curly braces.

cpojer commented on jest/setup.js:
```diff
@@ -156,6 +156,24 @@ const mockNativeModules = {
+    getScheduledLocalNotifications: jest.fn(callback => process.nextTick(() => callback())),
+    requestPermissions: jest.fn(() => Promise.resolve({ alert: true, badge: true, sound: true })),
+    abandonPermissions: jest.fn(),
+    checkPermissions: jest.fn(callback => process.nextTick(() => callback({ alert: true, badge: true, sound: true }))),
```
here too

cpojer commented:
Yap, this makes sense. Seems like CI is currently broken. If you rebase and make it green, I'm happy to merge it.

ZauberNerd commented:
@cpojer thanks for the feedback, that was extremely fast!
I fixed the issues mentioned above and rebased against master - the CI is still failing and judging from the other recently opened PRs it seems like the CI has been failing for most of them.
I'll try again tomorrow.

ZauberNerd commented:
@cpojer looking at the commits in master (https://github.com/facebook/react-native/commits/master) and their CI status, I can make the following observations:
- https://github.com/facebook/react-native/commit/cb3b744d084bbb3c44ee53b9da987b8b93c37c37 is the last commit that still runs circleci
- https://github.com/facebook/react-native/commit/65f22a51905c26bf388a7c9d6710e3d875ad832a and 4 commits later we see the first failing CI status.
- between those 4 commits, there is one, that adds/modifies `BUCK` files: https://github.com/facebook/react-native/commit/9468c5a733174fc4a956ede4824e41cb6149c393#diff-eeb363dcbace8ea59b94691412c00078R33

And by looking at the error message / stack trace from circleCI and the modifications of the aforementioned `BUCK` file, it seems like these changes might be related (although now I'm in an area that I have no expertise in - I have never before used `BUCK`).
```
BUILD FAILED: Couldn't get dependency '//ReactCommon/jschelpers:jschelpers' of target '//ReactAndroid/src/main/jni/packagerconnection:jni':
Parse error for build file /home/ubuntu/react-native/ReactCommon/jschelpers/BUCK:
NameError: name 'THIS_IS_FBANDROID' is not defined
Call stack:
  File "/home/ubuntu/buck/build/classes/buck_parser/buck.py", line 1216, in process_with_diagnostics
    diagnostics=diagnostics)
  File "/home/ubuntu/buck/build/classes/buck_parser/buck.py", line 1105, in process
    os.path.join(self._project_root, path))
  File "/home/ubuntu/buck/build/classes/buck_parser/buck.py", line 1041, in _process_build_file
    return self._process(build_env, path, is_implicit_include=False)
  File "/home/ubuntu/buck/build/classes/buck_parser/buck.py", line 975, in _process
    exec(code, module.__dict__)
  File "/home/ubuntu/react-native/ReactCommon/jschelpers/BUCK", line 16
    if THIS_IS_FBANDROID:
```
